src/mitigation.c

```diff
@@ -54,18 +56,37 @@ void mitigation_add_resource(struct mitigation *m, const char *name, const char
 
 	r->resource = res;
 
-	list_append(&m->resources, &r->list_node);
+	// resources that have the attribute fallback will be added to a seperate list
+	if(fallback)
+		list_append(&m->fallback_resources, &r->list_node);
+	else
+		list_append(&m->resources, &r->list_node);
 }
 
 void mitigation_activate(struct mitigation *m)
 {
 	struct mitigation_resource *r;
 	struct list_node *node;
+	int use_fallback_resources;
+	int ret;
 
+	use_fallback_resources = 0;
 	for_list_node(&m->resources, node) {
 		r = list_entry(node, struct mitigation_resource, list_node);
 		resource_enable(r->resource);
-		resource_write_value(r->resource, r->target_value, strlen(r->target_value));
+		ret = resource_write_value(r->resource, r->target_value, strlen(r->target_value));
+		// for an error in resource_write_value, declare the need to use fallback_resources
+		if(ret)
+			use_fallback_resources = 1;
```

Comment:
So if any resources fail, they aren't disabled and ALL fallback resources are activated?  This behavior needs to be more straightforward.  Either there should be specific fallback resources for other resources, or there should be a fallback behavior for an entire mitigation.